src/reanimated2/hook/useAnimatedGestureHandler.ts

```diff
@@ -31,78 +17,108 @@ const EventType = {
   END: 5,
 };
 
-interface GestureHandlerNativeEvent {
-  handlerTag: number;
-  numberOfPointers: number;
-  state: (typeof EventType)[keyof typeof EventType];
-}
+type StateType = (typeof EVENT_TYPE)[keyof typeof EVENT_TYPE];
 
-export interface GestureHandlerEvent<T> extends __NativeEvent<T> {
-  nativeEvent: T;
+interface PropsUsedInUseAnimatedGestureHandler {
+  handlerTag?: number;
+  numberOfPointers?: number;
+  state?: StateType;
+  oldState?: StateType;
```

Comment:
Should this be 
```suggestion
  previousState?: StateType;
```

Reply:
I see we use oldState a lot in our app. Is this previous state or is it even older?

Reply:
This is coming from `react-native-gesture-handler` https://github.com/software-mansion/react-native-gesture-handler/blob/main/src/handlers/gestureHandlerCommon.ts#L51, therefore we shouldn't rename it. `previousState` is also something, but it's not the case here (it's something different). Also keep in mind that `useAnimatedGestureHandler` is deprecated API and I we don't really want to bother that much about having it pretty.